### DISCUSSION TITLE
Magneto optical drive

### DISCUSSION THREAD

**Discussion by Teo290669** [BlueSCSI/BlueSCSI-v2#138] [General]
I'll try to ask you this question even though I think what I'm about to ask probably won't be possible:
Can bluescsi do what this product does?

https://www.cf2scsi.com/index.php?option=com_virtuemart&view=productdetails&virtuemart_product_id=21&Itemid=482

**Comment by erichelgeson** [comment]
I dont see why not - you can see on the Usage page we emulate MO drives https://bluescsi.com/docs/Usage#Usage

If your device is looking for a particular vendor you can set that in the ini file https://ini.bluescsi.com/

If there is a specific thing you're looking for then I could address that question directly - but in general I see no reason why not.

**Reply by Teo290669** [reply]
I think I understand that bluescsi is not able to read and write directly on the SD card but only via image files managed by appropriate software. Going into detail, what I would like to use the blues scsi for (which I have already purchased) is to replace a magneto optical drive that I use on a knitting machine which has its own proprietary hardware and software but nevertheless is connected to this system and therefore It manages a common SCSI magneto optical drive with which I use FAT 32 formatted magneto optical disks.

**Reply by erichelgeson** [reply]
BlueSCSI _is_ able to read and write directly to the SD card - just dont put any images on it. Since it is FAT32 you just need to setup the ini as an MO and place no images on the sd card. Give it a try and report back.

**Reply by Teo290669** [reply]
Ok, I'll try

**Reply by Teo290669** [reply]
I found that magneto optical disks are formatted fat16 but they also work if formatted fat 32/1Kb, so I formatted the SD card with this format.
I created a bluescsi.ini file containing:
[SCSI]
System=Generic
Type=4 ; Mag-optical
Unfortunately it didn't work, the system doesn't read anything.
Consider that the knitting machine in question was produced in the first half of the 90s.
I really hope it can work in my case so I'm attaching a photo of the reader using the machine.
![6c6598f3-5bcb-41bf-9b8a-7f3b30c88d23](https://github.com/BlueSCSI/BlueSCSI-v2/assets/166913563/420b1e83-b39e-465d-9a01-f04bc64a6be7)

**Reply by erichelgeson** [reply]
Thanks for the info. `Type` is a per-device setting (not the top level `[SCSI]` block) - and I see now that the `Type` is fixed to hdd when using raw mode. I can push a patch to allow this so you can test. I'll put the link here when it's ready.

**Reply by Teo290669** [reply]
Ok, thanks

**Reply by erichelgeson** [reply]
Download the "BlueSCSI Binaries" in this test build to try https://github.com/BlueSCSI/BlueSCSI-v2/actions/runs/8766101200

Though I'd strongly suggest you continue testing and debug with the "normal" Drive image file - You'll be able to test things without waiting for me. Once it is seen and working I can work back for options in raw mode.

**Reply by Teo290669** [reply]
ok, I'll try it and let you know as soon as possible

**Reply by Teo290669** [reply]
Nothing works yet. I think I have understood that the bluescsi is not recognized at all by the knitting machine because it behaves in the same way even when I have the scsi cable disconnected.
If it helps, this is a photo of the scsi controller board of the knitting machine.
[log.txt](https://github.com/BlueSCSI/BlueSCSI-v2/files/15066335/log.txt)
![IMG_3706](https://github.com/BlueSCSI/BlueSCSI-v2/assets/166913563/11bde041-cc8e-4d0c-91f5-cd11933a6950)
![IMG_3707](https://github.com/BlueSCSI/BlueSCSI-v2/assets/166913563/d26cbc88-62d4-4ee9-8993-2e586f81a1f7)

**Reply by erichelgeson** [reply]
As suggested - Please just try a MO file image (blank file of any size, named `MO1 test.hda` (for ID 1), and turn on debug log and attach that log.

**Reply by Teo290669** [reply]
Ok

**Comment by Teo290669** [comment]
I created the empty MO1 test.hda file using the string "dd if=/dev/zero of=example.hda bs=1M count=500" from the Mac terminal.
The bluescsi.ini file I created is as follows:
[SCSI]
System=Generic
Debug=1 ; Hon
Type=4 ; Mag-optical
Then I attach the generated log file

[log.txt](https://github.com/BlueSCSI/BlueSCSI-v2/files/15079443/log.txt)

**Reply by erichelgeson** [reply]
Awesome, now attach it to your device and try to format it. I see no SCSI commands logged so if you did hook it up your device didnt try to communicate with it yet. 

Note you do not need Type=4 as the file name `MO...` does that already.

**Reply by Teo290669** [reply]
Of course I tried to connect it but the result is just what you see.
I would add that the knitting machine is not able to format the magneto optical disk, it can only read and write, I obtain the formatting via a USB magneto optical disk reader connected to my PC

**Reply by erichelgeson** [reply]
Your machine is not sending any SCSI commands or bus resets at all then. I'd expect to see at least one bus reset, start/stop, or inquiry. Check to ensure your bus is properly terminated. Look at the manual for your machine to see if the MO has to be on a certain ID or something. If your host isn't sending commands to the BlueSCSI we can't debug any further without knowing why it's not.

**Reply by Teo290669** [reply]
Eric, thank you for your support, as I said at the beginning it will be very difficult because the knitting machine has its own system. I treasure your information and now I will try to understand what is preventing the system from connecting to the bluescsi. If I succeed I'll let you know.
